src/components/settings-table/dataset-table.js

```diff
@@ -1,126 +1,65 @@
-import React from 'react'
-
-import Table from '@material-ui/core/Table'
-import TableBody from '@material-ui/core/TableBody'
-import TableCell from '@material-ui/core/TableCell'
-import TableHead from '@material-ui/core/TableHead'
-import TableRow from '@material-ui/core/TableRow'
+import React, { useEffect, useState } from 'react'
+import Wrapper from '../wrapper'
 import DataSetTableRow from './dataset-table-row'
-
-import i18n from '@dhis2/d2-i18n'
-import dataTableStyles from '../../styles/DataTable.module.css'
 import { periodTypeConstants } from '../../constants/data-set-settings'
 
-class DataSetTable extends React.Component {
-    constructor(props) {
-        super(props)
-
-        this.state = {
-            dataSetSelected: this.props.dataSetSelected,
-            dataSetFilter: '',
-            periodType: '',
-            defaultValue: '',
-            selected: false,
-        }
-    }
+const DataSetTable = props => {
+    const [dataSetSelected, setDataset] = useState(props.dataSetSelected)
+    const [dataSetData, setData] = useState(props.data)
+    const [periodType, setPeriodType] = useState('')
+    const [defaultValue, setDefault] = useState('')
+    const [dataSetFilter, setDataSetFiltered] = useState('')
+    const [selected, setSelected] = useState(false)
 
-    checkPeriodType = () => {
-        if (this.props.dataSetSelected !== '') {
-            const dataSetSelected = this.props.completeListOptions.filter(
-                dataSet => dataSet.id === this.props.dataSetSelected
+    const checkPeriodType = edit => {
+        if (props.dataSetSelected !== '') {
+            const dataSetSelected = props.completeListOptions.filter(
+                dataSet => dataSet.id === props.dataSetSelected
             )
-            this.dataSetFilter = dataSetSelected[0]
+            if (!edit) {
+                props.states.periodDSDownload =
+                    periodTypeConstants[dataSetSelected[0].periodType].default
```

Comment:
This will directly mutate a deeply-nested value in `props` which **will update that value without rerendering any components**.  Mutation should always happen in a function which calls `setState` (perhaps in a component higher up the tree)

Reply:
I've moved this mutation to `dataSet-settings` thanks for the advice.